components/dashboard/src/components/UsageBasedBillingConfig.tsx

```diff
@@ -45,30 +45,34 @@ export default function UsageBasedBillingConfig({ attributionId }: Props) {
 
     const localStorageKey = `pendingStripeSubscriptionFor${attributionId}`;
     const now = dayjs().utc(true);
-    const billingPeriodFrom = now.startOf("month");
-    const billingPeriodTo = now.endOf("month");
+    const [billingCycleFrom, setBillingCycleFrom] = useState<dayjs.Dayjs>(now.startOf("month"));
+    const [billingCycleTo, setBillingCycleTo] = useState<dayjs.Dayjs>(now.endOf("month"));
+
+    const refreshSubscriptionDetails = async (attributionId: string) => {
+        setStripeSubscriptionId(undefined);
+        setIsLoadingStripeSubscription(true);
+        try {
+            const [subscriptionId, costCenter] = await Promise.all([
+                getGitpodService().server.findStripeSubscriptionId(attributionId),
+                getGitpodService().server.getCostCenter(attributionId),
+            ]);
+            setStripeSubscriptionId(subscriptionId);
+            setUsageLimit(costCenter?.spendingLimit);
+            setBillingCycleFrom(dayjs(costCenter?.billingCycleStart || now.startOf("month")).utc(true));
+            setBillingCycleTo(dayjs(costCenter?.nextBillingTime || now.endOf("month")).utc(true));
```

Comment:
Technically this needs to also be `.endOf('day')` because now it points to the start of the last day of the month. Not an issues since we ignore times here and when fetching the usage we use `now`.

Reply:
Sorry, I don't think I see the problem you're referring to here. Are you not seeing "last millisecond of the month" on your side?

<img width="385" alt="Screenshot 2022-11-14 at 08 46 23" src="https://user-images.githubusercontent.com/599268/201603639-6528f5cc-96a7-41a9-ad24-5a938739dc50.png">

Reply:
Oh, true. I didn't expect the `endOf('month')` to do the right thing already. But it makes sense. 👍